analyzers/src/SonarAnalyzer.Common/Rules/PartCreationPolicyShouldBeUsedWithExportAttributeBase.cs

```diff
@@ -44,30 +43,26 @@ protected void AnalyzeNode(SyntaxNodeAnalysisContext c)
                 return;
             }
 
-            var classDeclaration = GetClassDeclaration(attribute);
-            if (classDeclaration == null)
+            var declaration = GetTypeDeclaration(attribute);
+            if (declaration == null)
             {
                 return;
             }
 
-            var classSymbol = c.SemanticModel.GetDeclaredSymbol(classDeclaration) as ITypeSymbol;
-            if (classSymbol == null ||
-                classSymbol.AnyAttributeDerivesFrom(KnownType.System_ComponentModel_Composition_ExportAttribute) ||
-                classSymbol.GetSelfAndBaseTypes()
-                    .Union(classSymbol.AllInterfaces)
-                    .Any(s => s.AnyAttributeDerivesFrom(KnownType.System_ComponentModel_Composition_InheritedExportAttribute)))
+            if (!(c.SemanticModel.GetDeclaredSymbol(declaration) is ITypeSymbol symbol)
+                || symbol.AnyAttributeDerivesFrom(KnownType.System_ComponentModel_Composition_ExportAttribute)
+                || symbol.GetSelfAndBaseTypes()
+                         .Union(symbol.AllInterfaces)
+                         .Any(s => s.AnyAttributeDerivesFrom(KnownType.System_ComponentModel_Composition_InheritedExportAttribute)))
             {
                 return;
             }
 
-            c.ReportDiagnosticWhenActive(
-                Diagnostic.Create(
-                    SupportedDiagnostics[0],
-                    attribute.GetLocation()));
+            c.ReportDiagnosticWhenActive(Diagnostic.Create(SupportedDiagnostics[0], attribute.GetLocation()));
 
             bool IsPartCreationPolicyAttribute(TAttributeSyntax attributeSyntax) =>
-                c.SemanticModel.GetSymbolInfo(attributeSyntax).Symbol is IMethodSymbol attributeSymbol &&
-                attributeSymbol.ContainingType.Is(KnownType.System_ComponentModel_Composition_PartCreationPolicyAttribute);
+                c.SemanticModel.GetSymbolInfo(attributeSyntax).Symbol is IMethodSymbol attributeSymbol
```

Comment:
AFAIK _c.SemanticModel.GetSymbolInfo(c.Node).Symbol_ returns the same thing we already have in _c.ContainingSymbol_

Reply:
In this case no. `c.ContainingSymbol` for this attribute is the compilation unit